lib/sidekiq/grouping/batch.rb

```diff
@@ -117,8 +117,17 @@ def all
           redis = Sidekiq::Grouping::Redis.new
 
           redis.batches.map do |name|
-            new(*extract_worker_klass_and_queue(name))
-          end
+            if Sidekiq::Grouping::Config.reliable
+              begin
+                klass, queue = extract_worker_klass_and_queue(name)
+                Sidekiq::Grouping::ReliableBatch.new(klass, queue)
+              rescue NameError
```

Comment:
What are we rescuing from here? I don't see anything that would raise a `NameError`.

Reply:
This covers the case where a worker class with old batch jobs has been deleted. It no longer exists to be constantized but had batched jobs in the past and gets picked up by the query.

Reply:
I'll be honest, I found that case by mistake :)

Reply:
That makes sense, but does contantize happen inside this begin block? I only see the `extract_worker_klass_and_queue`, and the Batch initializer, neither of which appear to constantize the job name. I see that constantize is called when getting options, but we're no longer checking job options here. Maybe this is a leftover from when you had the `batch_reliable` option?

Reply:
I think you're right that it's a leftover now that I look around it. Good find.